Source/ACE/Entity/Container.cs

```diff
@@ -51,14 +52,54 @@ public Container(AceObject aceObject, ObjectGuid guid)
         {
         }
 
+        public virtual void ContainerPlacement(WorldObject inventoryItem)
+        {
+            var removedItem = inventory.Remove(inventoryItem.Guid);
+            var invList = inventory.OrderBy(o => o.Value.Placement);
+            uint place = 0;
+            AceObject saveableCopy;
+            foreach (var keyValuePair in invList)
+            {
+                if (place == inventoryItem.Placement)
+                    place++;
+
+                if (keyValuePair.Value.Placement != place)
+                {
+                    keyValuePair.Value.Placement = place;
+                    saveableCopy = keyValuePair.Value.SnapShotOfAceObject();
+                    saveableCopy.ClearDirtyFlags();
+                    saveableCopy.Placement = place;
+                    DatabaseManager.Shard.SaveObject(saveableCopy, null);
```

Comment:
Going to put this save in a #if directive it makes it easy to debug if it persists right away.   I will hook this in with character save which is how it will work.